### PR TITLE
Add scDblFinder script fro 10x wf

### PR DESCRIPTION
**Pull request by AntoniaChroni**
## Description

This is a notebook to use [ScDblFinder](https://bioconductor.org/packages/devel/bioc/vignettes/scDblFinder/inst/doc/scDblFinder.html) method for identifying doublets in single-cell data. 

 
- [ ] Replace scrublet with [scDblFinder](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02136-7) as described in # [issue2149](https://github.com/d3b-center/bixu-tracker/issues/2149#event-10682037261)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] [As solo tool](https://cavatica.sbgenomics.com/u/d3b-bixu/kf-dev-single-cell-rna/tasks/dedcd75f-95a7-44f3-959d-4febe0b44b05/)
- [ ] [In refinement wf](https://cavatica.sbgenomics.com/u/d3b-bixu/kf-dev-single-cell-rna/tasks/1817e047-b711-47ff-a5e6-2918bedde495/)

**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have committed any related changes to the PR

### CODE REVIEW

**Review by migbro — CHANGES_REQUESTED** [review]
Started a local run. Got a weird error upon local testing:
```
processing file: scDblFinder.Rmd
  |.....                             |  15% [load-library]                     System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
  |................                  |  46% [process-seurat-obj-convert-to-sce]
Quitting from lines 82-102 [process-seurat-obj-convert-to-sce] (scDblFinder.Rmd)
Error in `gzfile()`:
! invalid 'description' argument
Backtrace:
 1. base::readRDS(data_dir)
 2. base::gzfile(file, "rb")
```

**Comment by AntoniaChroni** [comment]
> Started a local run. Got a weird error upon local testing:
> 
> ```
> processing file: scDblFinder.Rmd
>   |.....                             |  15% [load-library]                     System has not been booted with systemd as init system (PID 1). Can't operate.
> Failed to connect to bus: Host is down
>   |................                  |  46% [process-seurat-obj-convert-to-sce]
> Quitting from lines 82-102 [process-seurat-obj-convert-to-sce] (scDblFinder.Rmd)
> Error in `gzfile()`:
> ! invalid 'description' argument
> Backtrace:
>  1. base::readRDS(data_dir)
>  2. base::gzfile(file, "rb")
> ```

As discussed this error is probably related to the wrong input file parsing. The correct one is the raw Seurat object: `*seurat_obj_raw.rds`

**Review by migbro — APPROVED** [review]
Looks good to me! I'll add to our code review for this week, and build off this on my other branch for the filter script update sand new cwl tool

File: scripts/scDblFinder.Rmd [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  library(Seurat)
+  library(scater)
+  library(future)
+  library(tidyverse)
+  library(grid)
+})
+
+# # evaluate Seurat R expressions asynchronously when possible (such as ScaleData) using future package
+plan("multisession", workers = 4)
+options(future.globals.maxSize = 64000 * 1024^2) # for 64 Gb RAM
```
Comment (dmiller15) [inline scripts/scDblFinder.Rmd:43]: These should probably be tied to the actual CPU and RAM values being passed to the software.

File: tools/scdblfinder.cwl [hunk 2]
```diff
@@ -14,7 +14,7 @@ requirements:
         entry:
           $include: ../scripts/scDblFinder.Rmd
   - class: ResourceRequirement
-    coresMin: 8
+    coresMin: $(inputs.cpus)
     ramMin: 16000
```
Comment (dmiller15) [inline tools/scdblfinder.cwl:18]: Can this be made a variable as well?

**Review by dmiller15 — APPROVED** [review]
Other than the last comment, looks good.

File: tools/scdblfinder.cwl (hunk 2 above)
Comment (migbro) [reply tools/scdblfinder.cwl:18]: Gah! That was an oversight. Will update before merge

**Pull request merged**